hack/tools/logcheck/main.go

```diff
@@ -27,14 +28,26 @@ import (
 	"golang.org/x/tools/go/analysis/singlechecker"
 )
 
+var logcheckFlags = flag.NewFlagSet("", flag.ExitOnError)
+var migrationFlagHelpText = `Migration test is done by explicitly passing -migration flag to
+logcheck. It runs correctness test but reports an error when
+unstructured logging function is used in a migrated package.
+
+Correctness test is run by default to check for correct usage
+of structured logging patterns. It reports an error if structured
+logging function is used incorrectly. It does not report an error
+when structured logging function is not used.`
+var migration = logcheckFlags.Bool("migration", false, migrationFlagHelpText)
```

Comment:
I think we should not name flags based on how we plan to use them, but what they really do. This is because purpose of this flag can change/get outdated, but what it does will not. As some point we might use this feature for something else than migration. We might end up with more flags, but that's ok.

Instead of "--migration", what about "allow-unstructured"?

Reply:
That's a great suggestion. Changed to "allow-unstructured".